auslib/dockerflow.py

```diff
@@ -26,8 +28,12 @@ def heartbeat():
         """Per the Dockerflow spec:
         Respond to /__heartbeat__ with a HTTP 200 or 5xx on error. This should
         depend on services like the database to also ensure they are healthy."""
-        database_entry_value = heartbeat_database_fn(dbo)
-        return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        try:
+            database_entry_value = heartbeat_database_fn(dbo)
+            return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        except Exception as e:
+            logging.exception(e)
+            return Response(status=502)
```

Comment:
You're on the right track here, but remember the bug asks for "useful information", so you'll want to provide the "response" arg to Response as well. Perhaps something similar to https://github.com/mozilla/balrog/blob/8f8057168b0ab13177468cac93b85af4cfcb2472/auslib/admin/views/permissions.py#L89 ?